src/pudl/metadata/classes.py

```diff
@@ -1470,15 +1498,19 @@ def to_sql(
             f.to_sql(
                 check_types=check_types,
                 check_values=check_values,
+                dialect=dialect,
             )
             for f in self.schema.fields
         ]
         constraints = []
         if self.schema.primary_key:
             constraints.append(sa.PrimaryKeyConstraint(*self.schema.primary_key))
-        for key in self.schema.foreign_keys:
-            constraints.append(key.to_sql())
-        return sa.Table(self.name, metadata, *columns, *constraints)
+        if check_foreign_keys:
+            for key in self.schema.foreign_keys:
+                constraints.append(key.to_sql())
+        return sa.Table(
+            self.name, metadata, *columns, *constraints, comment=self.description
+        )
```

Comment:
Previously the comment field on the Table was not being set, so there were no Table level descriptions.